src/storage/Storage.js

```diff
@@ -0,0 +1,21 @@
+class Storage {
```

Comment:
Thinking out loud here a bit. what other concrete implementations can you see surfacing for `Storage`?
* `SessionStorage`
* `IndexedDBStorage` orr `PersistentIfLuckyStorage`
* and then flavor of browser db as lib built on top of `IndexedDB` e.g. level

Reply:
theoretically it could even be something like `DWNStorage` or whatnot (which is why i made everything `async` from the getgo)